cpp/memilio/epidemiology/state_age_function.h

```diff
@@ -0,0 +1,464 @@
+ * 
+ * Example from IDE-SECIR model:
+ *
+ * ExponentialDecay expdecay(1.0);
+ * StateAgeFunctionWrapper prob(expdecay);
+ * model.parameters.set<mio::isecir::TransmissionProbabilityOnContact>(prob);
+ *
+ */
+struct StateAgeFunctionWrapper {
+
+    /**
+     * @brief Constructs a new StateAgeFunctionWrapper object
+     * 
+     * @param[in] init_function specifies the initial function.
+     */
+    StateAgeFunctionWrapper(StateAgeFunction& init_function)
+        : m_function(init_function.clone())
+    {
+    }
+
+    /**
+     * @brief Copy constructor. 
+     */
+    StateAgeFunctionWrapper(StateAgeFunctionWrapper const& other)
+        : m_function(other.m_function->clone())
+    {
+    }
+
+    /**
+     * @brief Move constructor. 
+     */
+    StateAgeFunctionWrapper(StateAgeFunctionWrapper&& other) = default;
+
+    /**
+     * @brief Copy assignment. 
+     */
+    StateAgeFunctionWrapper& operator=(StateAgeFunctionWrapper const& other)
+    {
+        m_function = other.m_function->clone();
+        return *this;
+    }
+
+    /**
+     * @brief Move assignment. 
+     */
+    StateAgeFunctionWrapper& operator=(StateAgeFunctionWrapper&& other) = default;
+
+    /**
+     * @brief Destructor. 
+     */
+    ~StateAgeFunctionWrapper() = default;
+
+    /**
+     * @brief Comparison operator. 
+     */
+    bool operator==(const StateAgeFunctionWrapper& other) const
+    {
+        return (m_function->get_state_age_function_type() == other.get_state_age_function_type() &&
+                m_function->get_parameter() == other.get_parameter());
+    }
+
+    /**
+     * @brief Set the StateAgeFunction object
+     *
+     * @param[in] new_function function that we want to set member m_function to.
+     */
+    void set_state_age_function(StateAgeFunction& new_function)
+    {
+        m_function = new_function.clone();
+    }
+
+    /**
+     * @brief Get type of StateAgeFunction, i.e. which derived class is used.
+     *
+     * @param[out] string 
+     */
+    std::string get_state_age_function_type() const
+    {
+        return m_function->get_state_age_function_type();
+    }
+
+    /**
+     * @brief Accesses eval of m_function.
+     *
+     * @param[in] state_age Time at which the function is evaluated.
+     * @return ScalarType evaluation of the function at state_age. 
+     */
+    ScalarType eval(ScalarType state_age) const
+    {
+        return m_function->eval(state_age);
+    }
+
+    /**
+     * @brief Get the m_parameter object of m_function.
+     * 
+     * @return ScalarType 
+     */
+    ScalarType get_parameter() const
+    {
+        return m_function->get_parameter();
+    }
+
+    /**
+     * @brief Set the m_parameter object of m_function. 
+     * 
+     * @param[in] new_parameter that determines new function parameter
+     */
+    void set_parameter(ScalarType new_parameter)
+    {
+        m_function->set_parameter(new_parameter);
+    }
+
+    ScalarType get_max_support(ScalarType dt) const
```

Comment:
If we always call via this wrapper, we should have a potential second argument `tol` here too, shouldnt we?

Reply:
We could also make tol a member of StateAgeFunction. I just pushed a version where I did that and where I also separated computing the max support and the getter. There is one thing that isn't working perfectly there yet. We should discuss if this separation makes sense.